openmls/src/group/tests/test_past_secrets.rs

```diff
@@ -0,0 +1,231 @@
+//! This module contains tests regarding the use of [`SecretTreeStore`] in [`ManagedGroup`]
+
+use openmls_rust_crypto::OpenMlsRustCrypto;
+use openmls_traits::{key_store::OpenMlsKeyStore, OpenMlsCryptoProvider};
+
+use rstest::*;
+use rstest_reuse::{self, *};
+
+use crate::{
+    ciphersuite::{Ciphersuite, CiphersuiteName},
+    config::Config,
+    credentials::{CredentialBundle, CredentialType},
+    framing::{MlsCiphertextError, ProcessedMessage},
+    group::{
+        GroupId, ManagedGroup, ManagedGroupConfig, ManagedGroupError, MlsGroupError, WireFormat,
+    },
+    key_packages::KeyPackageBundle,
+    tree::secret_tree::SecretTreeError,
+};
+
+#[apply(ciphersuites_and_backends)]
+fn test_past_secrets_in_group(
```

Comment:
Don't we have the test framework for these lengthy group setups? It should work pretty well here.